src/lib/hooks/useBlockNumber.tsx

```diff
@@ -28,9 +31,14 @@ export function useFastForwardBlockNumber(): (block: number) => void {
   return useBlockNumberContext().fastForward
 }
 
+export function useEthBlockNumber(): number | undefined {
+  return useBlockNumberContext().ethValue
+}
+
 export function BlockNumberProvider({ children }: { children: ReactNode }) {
   const { chainId: activeChainId, provider } = useWeb3React()
   const [{ chainId, block }, setChainBlock] = useState<{ chainId?: number; block?: number }>({ chainId: activeChainId })
+  const [ethBlockNumber, setEthBlockNumber] = useState<number | undefined>(undefined)
```

Comment:
I think you can incorporate this into the existing state, and that it shouldn't be separated. Can you augment the chainBlock state to include `mainnetBlock`?